From d3277d83a0118294d1101e3b22f53e686866aa57 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Fri, 16 Aug 2019 18:09:55 +0530
Subject: [PATCH 08/27] arm64: Increase NR_IRQS to a large number

commit 98df58c8874349fe287a41e7ed667721eb7022cb from
git@git.assembla.com:cavium/WindRiver.linux.git

Platforms with multiple SR-IOV capable PCI devices will
need large number of MSIX vectors. Hence increase
NR_IRQS to 64K, so that enabling MSIX for a device doesn't
fail due to shortage of IRQs in the system.

Change-Id: I6b36873d24a054e4691da0b0586a41328791d37a
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/14428
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/include/asm/irq.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/include/asm/irq.h b/arch/arm64/include/asm/irq.h
index a0fee6985e6a..af6c1ff1981f 100644
--- a/arch/arm64/include/asm/irq.h
+++ b/arch/arm64/include/asm/irq.h
@@ -6,6 +6,15 @@
 
 #include <asm-generic/irq.h>
 
+/* Platforms with multiple SR-IOV capable PCI devices will
+ * need large number of MSIX vectors, hence keep this number
+ * fairly high.
+ */
+#ifdef CONFIG_PCI_MSI
+#undef  NR_IRQS
+#define NR_IRQS        65536
+#endif
+
 struct pt_regs;
 
 extern void set_handle_irq(void (*handle_irq)(struct pt_regs *));
-- 
2.17.1

